posthog/clickhouse/property_groups.py

```diff
@@ -0,0 +1,72 @@
+from collections.abc import MutableMapping
+from dataclasses import dataclass
+
+
+@dataclass
+class PropertyGroupDefinition:
+    filter_expression: str
+    codec: str = "ZSTD(1)"
```

Comment:
This will eventually be extended for use on the read path so that a property key can be tested for membership in this property group. This is going to be a bit messy for a while since logic will need to be duplicated between the Python version used in the [PropertySwapper](https://github.com/PostHog/posthog/blob/00bea7f2f0aec09acb91cbab18fd571144df3466/posthog/hogql/transforms/property_types.py#L117) and the ClickHouse version used in the materialization expression. For the `feature_flags` group below, `"key like '$feature/%'"` will end up as `operator.startswith('$feature/')` or something. `custom` will be a little more complicated. It'd be nice if these could be unified in some way, but that's a bigger can of worms that doesn't need to be opened right now.